edgelet/iotedge/src/check/mod.rs

```diff
@@ -1501,6 +1601,98 @@ where
     Ok(output.stdout)
 }
 
+// Resolves the given `ToSocketAddrs`, then connects to the first address via TCP and completes a TLS handshake.
+//
+// `tls_hostname` is used for SNI validation and certificate hostname validation.
+//
+// `hostname_display` is used for the error messages.
+fn resolve_and_tls_handshake(
+    to_socket_addrs: &impl std::net::ToSocketAddrs,
+    tls_hostname: &str,
+    hostname_display: &str,
+) -> Result<(), failure::Error> {
+    let host_addr = to_socket_addrs
+        .to_socket_addrs()
+        .with_context(|_| {
+            format!(
+                "Could not connect to {} : could not resolve hostname",
+                hostname_display,
+            )
+        })?
+        .next()
+        .ok_or_else(|| {
+            Context::new(format!(
+                "Could not connect to {} : could not resolve hostname: no addresses found",
+                hostname_display,
+            ))
+        })?;
+
+    let stream = TcpStream::connect_timeout(&host_addr, std::time::Duration::from_secs(10))
+        .with_context(|_| format!("Could not connect to {}", hostname_display))?;
+
+    let tls_connector = native_tls::TlsConnector::new().with_context(|_| {
+        format!(
+            "Could not connect to {} : could not create TLS connector",
+            hostname_display,
+        )
+    })?;
+
+    let _ = tls_connector
+        .connect(tls_hostname, stream)
+        .with_context(|_| {
+            format!(
+                "Could not connect to {} : could not complete TLS handshake",
+                hostname_display,
+            )
+        })?;
+
+    Ok(())
+}
+
+fn is_rfc_1035_valid(name: &str) -> bool {
+    if name.len() > 255 {
```

Comment:
Don't we also need to check for `name.len() > 0`?

Reply:
I don't *think* it's possible to call this function with `name == ""` since it would also require `gethostname` to return the empty string, but it doesn't hurt.

Reply:
This is a private function so it's fine to make that assumption, although in that case it might be good to assert it somehow.

Reply:
Done.